src/main/java/Duke.java

```diff
@@ -1,10 +1,137 @@
+/**
+ * Solution below adapted from
+ * https://github.com/nus-cs2113-AY2021S1/contacts/blob/master/src/main/java/Contacts1.java
+ */
+
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final int MAX_NUMBER_OF_TASKS = 100;
+    private static Task[] tasks = new Task[MAX_NUMBER_OF_TASKS];
+    private static int taskNumber = 0;
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+        while (true) {
+            String command = getUserCommand();
+            processUserCommand(command);
+        }
+
+    }
+
+    private static void processUserCommand(String command) {
+        if (command.equals("bye")) {
+            printExitMessage();
+            exitDuke();
+        } else if (command.equals("list")) {
+            printTaskList();
+        } else if (command.startsWith("done")) {
+            Task finishedTask = markTaskAsDone(command);
+            printFinishedTask(finishedTask);
+        } else {
+            Task newTask = addNewTask(command);
+            printNewTask(newTask);
+        }
+    }
+
+    private static void printNewTask(Task newTask) {
+        System.out.println("____________________________________________________________");
+        System.out.println("Got it. I've added this task:" + System.lineSeparator() + "  " + newTask
+                + System.lineSeparator() + "Now you have " + (taskNumber) + " task(s) in the list.");
+        System.out.println("____________________________________________________________");
+    }
+
+    private static Task addNewTask(String command) {
+        Task newTask;
+        if (command.startsWith("todo")) {
+            newTask = createTodo(command);
+        } else if (command.startsWith("deadline")) {
+            newTask = createDeadline(command);
+        } else if (command.startsWith("event")) {
+            newTask = createEvent(command);
+        } else {
+            newTask = createTask(command);
+        }
+        tasks[taskNumber] = newTask;
+        taskNumber++;
+        return newTask;
+    }
```

Comment:
Well done :) I think it is good that you have abstracted the individual commands into their own function instead of putting them all in this function.